controlplane/kubeadm/controllers/helpers.go

```diff
@@ -256,16 +255,36 @@ func (r *KubeadmControlPlaneReconciler) generateMachine(ctx context.Context, kcp
 	return nil
 }
 
-// machinesNeedingRollout return a list of machines that need to be rolled out.
-func (r *KubeadmControlPlaneReconciler) machinesNeedingRollout(ctx context.Context, c *internal.ControlPlane) internal.FilterableMachineCollection {
-	// Ignore machines to be deleted.
-	machines := c.Machines.Filter(machinefilters.Not(machinefilters.HasDeletionTimestamp))
-
-	// Return machines if they are scheduled for rollout or if with an outdated configuration.
-	return machines.AnyFilter(
-		// Machines that are scheduled for rollout (KCP.Spec.UpgradeAfter set, the UpgradeAfter deadline is expired, and the machine was created before the deadline).
-		machinefilters.ShouldRolloutAfter(c.KCP.Spec.UpgradeAfter),
-		// Machines that do not match with KCP config.
-		machinefilters.Not(machinefilters.MatchesKCPConfiguration(ctx, r.Client, c.KCP)),
-	)
+func (r *KubeadmControlPlaneReconciler) getInfraObjects(ctx context.Context, machines internal.FilterableMachineCollection) (map[string]*unstructured.Unstructured, error) {
```

Comment:
nit: I think Resources is a bit more in line with common terminology
```suggestion
func (r *KubeadmControlPlaneReconciler) getInfraResources(ctx context.Context, machines internal.FilterableMachineCollection) (map[string]*unstructured.Unstructured, error) {
```